src/test/spec/initial_dns_seedlist_discovery.rs

```diff
@@ -68,44 +80,66 @@ async fn run() {
 
         assert_eq!(expected_seeds, actual_seeds,);
 
-        // The spec tests use two fields to compare against for the authentication database. In the
-        // `parsed_options` field, there is a `db` field which is populated with the database
-        // between the `/` and the querystring of the URI, and in the `options` field, there is an
-        // `authSource` field that is populated with whatever the driver should resolve `authSource`
-        // to from both the URI and the TXT options. Because we don't keep track of where we found
-        // the auth source in ClientOptions and the point of testing this behavior in this spec is
-        // to ensure that the right database is chosen based on both the URI and TXT options, we
-        // just determine what that should be and stick that in all of the options parsed from the
-        // spec JSON.
-        let resolved_db = test_file
-            .options
-            .as_ref()
-            .and_then(|opts| opts.auth_source.clone())
-            .or_else(|| {
-                test_file
-                    .parsed_options
-                    .as_ref()
-                    .and_then(|opts| opts.db.clone())
-            });
+        // "txt-record-with-overridden-ssl-option.json" requires SSL be disabled; see DRIVERS-1324.
+        let requires_tls = match test_file.options {
+            Some(ref options) => options.ssl,
+            None => true,
+        };
+        let client = TestClient::new().await;
+        if requires_tls == client.options.tls_options().is_some()
+            && client.is_replica_set()
+            && client.options.repl_set_name.as_deref() == Some("repl0")
+        {
+            // If the connection URI provides authentication information, manually create the user
+            // before connecting.
+            if let Some(ParsedOptions {
+                user: Some(ref user),
+                password: Some(ref pwd),
+                ref db,
+            }) = test_file.parsed_options
+            {
+                client
```

Comment:
Authentication isn't required for monitoring threads to discover the topology, so I don't think it's necessary that we create these users.

Reply:
`TestClient::new()` sends a few commands via `execute_operation_on_session` which goes through the authentication process for URIs that specify a username/password pair; commenting out this step causes a SCRAM failure.